packages/cli/src/handlers/dbt/compile.ts

```diff
@@ -120,48 +121,7 @@ const getJoinedModelsRecursively = (
     );
 };
 
-export const compileModelsAndJoins = async (
-    loadManifestOpts: LoadManifestArgs,
-    options: DbtCompileOptions,
-) => {
-    // do initial compilation so we can get the list of models that are compiled after this command (e.g. selecting/excluding by tags)
-    await dbtCompile(options);
-
-    // If no models are explicitly selected or excluded, we don't need to explicitly find joined models
-    if (!options.select && !options.exclude) {
-        return;
-    }
-
-    // Load manifest and get all models
-    const manifest = await loadManifest(loadManifestOpts);
-    const allManifestModels = getModelsFromManifest(manifest);
-    const currCompiledModels = getCompiledModels(allManifestModels);
-
-    // Selected models and their joined models
-    const modelsToCompile = new Set(
-        currCompiledModels.reduce<string[]>((acc, model) => {
-            const joinedModelNames = getJoinedModelsRecursively(
-                model,
-                allManifestModels,
-                new Set(acc), // minimize recursion by passing already visited models in the current list
-            );
-            return [...acc, model.name, ...joinedModelNames];
-        }, []),
-    );
-
-    GlobalState.debug(
-        `> Models to compile: ${Array.from(modelsToCompile).join(', ')}`,
-    );
-
-    await dbtCompile({
-        ...options,
-        select: Array.from(modelsToCompile),
-    });
-};
-
-export const dbtList = async (
-    options: DbtCompileOptions,
-): Promise<string[]> => {
+async function dbtList(options: DbtCompileOptions): Promise<string[]> {
```

Comment:
Didn't change the `dbtList` method.
Since I moved the `dbtList` call inside `maybeCompileModelsAndJoins` I can to move it to be above it.